File: Common/Entities/Pager.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SampleProject.Common.Entities
{
    public class Pager
    {
        public int PageIndex { get; set; } = 1;
        public int PageSize { get; set; }
        public int FilterBy { get; set; }
        public int SortBy { get; set; }
        public int SortDirection { get; set; }
        public string FilterText { get; set; } = "";

        public string SortByField => ((SortFields)SortBy).ToString();
        public string SortDirectionField => ((SortDirection)SortDirection).ToString();
        public int SkipBy
        {
            get
            {
                return this.PageSize * (this.PageIndex - 1);
            }
        }
    }

}

File: FRCSPreparationPortal/FRCSPreparationPortalUI/src/app/entities/pager.ts
import { SortDirection } from "../shared/enums";

export class Pager {
    public constructor(init?: Partial<Pager>) {
        Object.assign(this, init);
    }

    selectItemsPerPage: number[] = [5, 10, 25, 50, 100];
    pageSize = 20;
    pageIndex: number = 1;
    allItemsLength: number = 0;
    sortBy: number;
    filterBy:number;
    filterText:string;
    sortDirection: number = SortDirection.Desc ;
}
